src/Microsoft.OpenApi/Models/OpenApiSchema.cs

```diff
@@ -354,6 +363,9 @@ public void SerializeAsV3WithoutReference(IOpenApiWriter writer)
             // properties
             writer.WriteOptionalMap(OpenApiConstants.Properties, Properties, (w, s) => s.SerializeAsV3(w));
 
+            // patternProperties
+            writer.WriteOptionalMap(OpenApiConstants.PatternProperties, PatternProperties, (w, s) => s.SerializeAsV3(w));
+
```

Comment:
Does this not also needed to be added in SerializeAsV2WithoutReference ?

Reply:
@darrelmiller patternProperties field is not supported in OpenAPI v2 (https://swagger.io/docs/specification/data-models/keywords/), so I didn't add it.

Reply:
It turns out, neither does OpenAPI v3, which would explain why we didn't implement it.  It is however supported in v3.1, so lets hold this PR until V3.1 which we should be working on in January.
https://spec.openapis.org/oas/v3.0.3#properties

Reply:
Sure, we will hold the PR until we start on v3.1